feature/deps.go

```diff
@@ -0,0 +1,109 @@
+package feature
+
+import (
+	"fmt"
+
+	"github.com/lightningnetwork/lnd/lnwire"
+)
+
+type (
+	// featureSet contains a set of feature bits.
+	featureSet map[lnwire.FeatureBit]struct{}
+
+	// checkedFeatures maps a feature bit to a boolean indicating if this
+	// features depdencies have already been verified. This allows us to
+	// short circuit verification if multiple features have common
+	// depdencies, or map traversal starts verifying from the bottom up.
+	checkedFeatures map[lnwire.FeatureBit]bool
+
+	// depDesc maps a features to its set of depdent features, which must
+	// also be present for the vector to be valid. This can be used to
+	// recursively check the dependency chain for features set in a feature
+	// vector.
+	depDesc map[lnwire.FeatureBit]featureSet
+)
+
+// ErrMissingFeatureDep is an error signaling that a transitive dependency in a
+// feature vector is not set properly.
+type ErrMissingFeatureDep struct {
+	dep lnwire.FeatureBit
+}
+
+// Error returns a human-readable description of the missing dep error.
+func (e ErrMissingFeatureDep) Error() string {
+	return fmt.Sprintf("missing feature: %v", e.dep)
+}
+
+// deps is the default set of dependencies for assigned feature bits. If a
+// feature is not present in the depDesc it is assumed to have no depdencies.
+var deps = depDesc{
+	lnwire.PaymentAddrOptional: {
+		lnwire.TLVOnionPayloadOptional: {},
+	},
+	lnwire.MPPOptional: {
+		lnwire.PaymentAddrOptional: {},
+	},
+}
+
+// ValidateDeps asserts that a feature vector sets all features and their
+// transitive dependencies properly. It assumes that the dependencies between
+// optional and required features are identical, e.g. if a feature is required
+// but its depdency is optional, that is sufficient.
+func ValidateDeps(fv *lnwire.FeatureVector) error {
+	features := fv.Features()
+	checked := initChecked(features)
+
+	return validateDeps(features, checked)
+}
```

Comment:
Can't we just do two if statements for now instead of this more advanced recursive bit flipping solution?